modules/azure/loadbalancer.go

```diff
@@ -0,0 +1,201 @@
+}
+
+// LoadBalancerExists indicates whether the specified Load Balancer exists.
+// This function would fail the test if there is an error.
+func LoadBalancerExists(t testing.TestingT, loadBalancerName string, resourceGroupName string, subscriptionID string) bool {
+	exists, err := LoadBalancerExistsE(loadBalancerName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// LoadBalancerExistsE indicates whether the specified Load Balancer exists.
+func LoadBalancerExistsE(loadBalancerName string, resourceGroupName string, subscriptionID string) (bool, error) {
+	_, err := GetLoadBalancerE(loadBalancerName, resourceGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// GetLoadBalancerConfigNames gets a list of the Frontend Configuration Names for the Load Balancer.
+// This function would fail the test if there is an error.
+func GetLoadBalancerConfigNames(t testing.TestingT, loadBalancerName string, resourceGroupName string, subscriptionID string) []string {
+	configName, err := GetLoadBalancerConfigNamesE(loadBalancerName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return configName
+}
+
+// GetLoadBalancerConfigNamesE gets a list of the Frontend Configuration Names for the Load Balancer.
+func GetLoadBalancerConfigNamesE(loadBalancerName string, resourceGroupName string, subscriptionID string) ([]string, error) {
+	var configNames []string
+
+	lb, err := GetLoadBalancerE(loadBalancerName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return configNames, err
+	}
+
+	lbProps := lb.LoadBalancerPropertiesFormat
+	feConfigs := *lbProps.FrontendIPConfigurations
+	if len(feConfigs) == 0 {
+		// No Frontend Configurations present
+		return configNames, NewNotFoundError("Frontend Config", "Any", loadBalancerName)
+	}
+
+	// Get the names of the Frontend IP Configurations present
+	for _, config := range feConfigs {
+		configNames = append(configNames, *config.Name)
+	}
+
+	return configNames, nil
+}
+
+// GetLoadBalancerFrontendConfig gets the IP address and specifies public or private for the specified Load Balancer.
+// This function would fail the test if there is an error.
+func GetLoadBalancerFrontendConfig(t testing.TestingT, feConfigName string, loadBalancerName string, resourceGroupName string, subscriptionID string) (ipAddress string, publicOrPrivate IPType) {
+	ipAddress, ipType, err := GetLoadBalancerFrontendConfigE(feConfigName, loadBalancerName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return ipAddress, ipType
+}
+
+// GetLoadBalancerFrontendConfigE gets the IP address and specifies public or private for the specified Load Balancer.
+func GetLoadBalancerFrontendConfigE(feConfigName string, loadBalancer01Name string, resourceGroupName string, subscriptionID string) (ipAddress string, publicOrPrivate IPType, err1 error) {
```

Comment:
func name should be GetLoadBalancerIPE as your not returning the whole frontend config.

Reply:
Updated